v19.1/cost-based-optimizer.md

```diff
@@ -328,6 +331,236 @@ $ cockroach sql --insecure --host=localhost --port=26259 --database=test -e 'EXP
+
+The instructions below assume that you are already familiar with:
+
+- How to [Start a local cluster](start-a-local-cluster.html).
+- The syntax for [assigning node locality when configuring replication zones](configure-replication-zones.html#descriptive-attributes-assigned-to-nodes).
+- Using [the built-in SQL client](use-the-built-in-sql-client.html).
+
+First, start 3 local nodes as shown below. Use the [`--locality`](start-a-node.html#locality) flag to put them each in a different region.
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach start --locality=region=us-east  --insecure --store=/tmp/node0 --listen-addr=localhost:26257 --http-port=8888  --join=localhost:26257,localhost:26258,localhost:26259 --background
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach start --locality=region=us-central   --insecure --store=/tmp/node1 --listen-addr=localhost:26258 --http-port=8889  --join=localhost:26257,localhost:26258,localhost:26259 --background
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach start --locality=region=us-west --insecure --store=/tmp/node2 --listen-addr=localhost:26259 --http-port=8890  --join=localhost:26257,localhost:26258,localhost:26259 --background
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach init --insecure --host=localhost --port=26257
+~~~
+
+From the SQL client, add your organization name and enterprise license:
+
+{% include copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING cluster.organization = 'FooCorp - Local Testing';
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING enterprise.license = 'xxxxx';
+~~~
+
+Create an authentication database and table:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE DATABASE if NOT EXISTS auth;
+> USE auth;
+
+> CREATE TABLE token (
+	token_id VARCHAR(100) NULL,
+	access_token VARCHAR(4000) NULL,
+	refresh_token VARCHAR(4000) NULL
+  );
+~~~
+
+Create the indexes for each region:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE INDEX token_id_west_idx ON token (token_id) STORING (access_token, refresh_token);
+
+> CREATE INDEX token_id_central_idx ON token (token_id) STORING (access_token, refresh_token);
+
+> CREATE INDEX token_id_east_idx ON token (token_id) STORING (access_token, refresh_token);
+~~~
+
+Enter zone configurations to distribute replicas across the cluster as follows:
+
+- For the "East" index, store 2/5 replicas in the East, 2/5 in Central, and 1/5 in the West. Further, prefer that the leaseholders for that index live in the East or, failing that, in the Central region.
+- Follow the same replica and leaseholder patterns for each of the Central and West regions.
+
+The idea is that e.g. `token_id_east_idx` will have sufficient replicas and its leaseholder stored in the East, so that if a query comes in that accesses the columns covered by that index from the East gateway node, the optimizer will select `token_id_east_idx` for fast reads.
```

Comment:
Might want to also mention that you're deliberately putting 2 replicas in the preferred region so that if one replica goes down, the leaseholder will still stay in that region.

Reply:
Updated to make this explicit.